cmd/commandfuncs.go

```diff
@@ -538,6 +538,37 @@ func cmdValidateConfig(fl Flags) (int, error) {
 	return caddy.ExitCodeSuccess, nil
 }
 
+func cmdFormatConfig(fl Flags) (int, error) {
+	formatCmdConfigFlag := fl.String("config")
+	formatCmdFormatterFlag := fl.String("formatter")
+	// Chose caddyfile by default
+	if formatCmdFormatterFlag == "" {
+		formatCmdFormatterFlag = "caddyfile"
+	}
+
+	input, err := ioutil.ReadFile(formatCmdConfigFlag)
+	if err != nil {
+		return caddy.ExitCodeFailedStartup,
+			fmt.Errorf("reading input file: %v", err)
+	}
+
+	formatter := caddyconfig.GetFormatter(formatCmdFormatterFlag)
+	if formatter == nil {
+		return caddy.ExitCodeFailedStartup,
+			fmt.Errorf("unrecognized config formatter: %s", formatCmdFormatterFlag)
+	}
+
+	output := formatter.Format(input)
+	// fmt.Print(string(output))
+
+	err = ioutil.WriteFile(formatCmdConfigFlag, output, 0644)
```

Comment:
I think the command must not mutate the input file. Dangerous and unexpected! Print to stdout instead. User can redirect to a file if they want. Or we can add a flag for it. This would be my first suggestion for a revision.

Reply:
FWIW I'd actually expect it to mutate the input. That's how most `fmt` things work. I feel like having a `--dry-run` option or similar would make more sense, which would just output to stdout.

Reply:
@francislavoie 

>  That's how most fmt things work.

Citation needed. 😁  (But they won't change my mind, I'm afraid.)

Here, I'll give you a counterexample: https://golang.org/cmd/gofmt/ -- does not overwrite the input.

Overwriting input files is risky, and I feel pretty strongly that we shouldn't do that.

If you want to do that, it's easy with my proposed change:

```
caddy fmt > Caddyfile
```

This also makes the command more convenient for scripting and other use.

Plus, I've found that it's quite common in practice that the caddy process does not have write privileges on its input config file.

The sweet spot for this feature is potential editor integration in the future.

Reply:
> Here, I'll give you a counterexample: golang.org/cmd/gofmt -- does not overwrite the input.

Could've sworn it did. Haven't played with Go in a while so 🤷‍♂ I stand corrected

I don't feel super strongly about this point, so I concede 😄

Reply:
I'm with Matt here

Reply:
> caddy fmt > Caddyfile

So I think I should also default `--config` to take `Caddyfile` so we can use it this way.

And is it better to accept file as an argument or flag?

Reply:
FWIW `caddy fmt > Caddyfile` won't work because it will cause Caddyfile to be replaced with an empty file before `caddy fmt` runs. You can try it easily, the following will empty up file:

```
echo test > file
cat file > file
```

So you'd have to do it in a two step process of `caddy fmt > Caddyfile.tmp && mv Caddyfile.tmp Caddyfile`. Not a big deal though.

Reply:
Wow, I hate bash.

Reply:
Yeah, and so I added the `--write` option :D